OmniKit/PumpManager/PodCommsSession.swift

```
@@ -326,8 +324,34 @@ public class PodCommsSession {
         podState.updateFromStatusResponse(status)
         return status
     }
-    
-    public func insertCannula() throws -> TimeInterval {
+
+    // emits the specified beep type and sets the completion beep flags based on the specified confirmationBeep value
+    public func beepConfig(beepConfigType: BeepConfigType, confirmationBeeps: Bool) throws {
+        guard self.podState.fault == nil else {
+            return // skip if already faulted to avoid a Beep Config Command error response
+        }
+        let basalCompletionBeep: Bool = confirmationBeeps ? basalBeeps : false
+        let tempBasalCompletionBeep: Bool = confirmationBeeps ? tempBasalBeeps : false
+        let bolusCompletionBeep: Bool = confirmationBeeps ? bolusBeeps : false
+
+        let beepConfigCommand = BeepConfigCommand(beepConfigType: beepConfigType, basalCompletionBeep: basalCompletionBeep, tempBasalCompletionBeep: tempBasalCompletionBeep, bolusCompletionBeep: bolusCompletionBeep)
+        let statusResponse: StatusResponse = try send([beepConfigCommand])
+        podState.updateFromStatusResponse(statusResponse)
+    }
+
+    private func configureCompleted(confirmationBeeps: Bool, optionalPodAlarms: Bool) throws {
+        podState.setupProgress = .completed
+
+        if optionalPodAlarms {
+            // program optional Pod alarms, will beep upon completion if supplementaryBeeps && confirmationBeeps
+            try enableOptionalPodAlarms(confirmationBeeps: confirmationBeeps)
```

If these fail, the exceptions will be thrown back to insertCannula, and reported as an error to the user, even though the setupProgress is .completed.  The UI will offer a retry to the user, which will then fail, as the pod is not in the right state.  If you want to consider this successful, I'd recommend swallowing the exceptions for the beeps here.